### PR TITLE
Conduct code cleanup in the sandbox

### PR DESCRIPTION
**Pull request by mawiesne**
Change
-
- conducts code cleanup of some sandbox classes
- removes unnecessary duplicates
- modernizes some code style towards Java 8+
- adds missing break statements in loops to terminate earlier 
- simplifies some methods
- fixes problematic JavaDoc
- removes unused imports

Tasks
-
Thank you for contributing to Apache OpenNLP.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with OPENNLP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn clean install` at the root opennlp-sandbox folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file in opennlp-sandbox folder?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found in opennlp-sandbox folder?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by kinow — APPROVED** [review]

File: opennlp-similarity/src/main/java/opennlp/tools/similarity/apps/ContentGeneratorSupport.java [hunk 1]
```diff
@@ -77,9 +77,12 @@ public static List<String> buildSearchEngineQueryFromSentence(String sentence) {
 				boolean bAccept = true;
 				for (String w : qs) {
 					if (w.toLowerCase().equals(w)) // idf only two words then
-						// has to be person name,
-						// title or geolocation
+					// has to be person name,
+					// title or geolocation
+					{
 						bAccept = false;
+						break;
```
Comment (rzo1) [inline opennlp-similarity/src/main/java/opennlp/tools/similarity/apps/ContentGeneratorSupport.java:84]: If we break early, we can remove `bAccept` here? It is only used for a `continue` later, which is obsolete if we break early.

File: opennlp-similarity/src/main/java/opennlp/tools/similarity/apps/RelatedSentenceFinder.java [hunk 2]
```diff
@@ -217,9 +217,12 @@ public static List<String> buildSearchEngineQueryFromSentence(String sentence) {
 				boolean bAccept = true;
 				for (String w : qs) {
 					if (w.toLowerCase().equals(w)) // idf only two words then
-						// has to be person name,
-						// title or geolocation
+					// has to be person name,
+					// title or geolocation
+					{
 						bAccept = false;
```
Comment (rzo1) [inline opennlp-similarity/src/main/java/opennlp/tools/similarity/apps/RelatedSentenceFinder.java:223]: same as above

File: opennlp-similarity/src/main/java/opennlp/tools/parse_thicket/apps/MinedSentenceProcessor.java
[the remark this answers is not in the document]
Comment (mawiesne) [reply opennlp-similarity/src/main/java/opennlp/tools/parse_thicket/apps/MinedSentenceProcessor.java:1]: Unused.

[the remark this answers is not in the document]
Comment (mawiesne) [reply opennlp-similarity/src/main/java/opennlp/tools/parse_thicket/apps/MinedSentenceProcessor.java:1]: Even an unused duplicate / copy.

File: opennlp-similarity/src/main/java/opennlp/tools/similarity/apps/ContentGeneratorSupport.java (hunk 1 above)
Comment (rzo1) [reply opennlp-similarity/src/main/java/opennlp/tools/similarity/apps/ContentGeneratorSupport.java:84]: Missed the outer loop. Nevermind :-)

File: opennlp-similarity/src/main/java/opennlp/tools/similarity/apps/ContentGenerator.java [hunk 3]
```diff
@@ -146,9 +146,12 @@ public static List<String> buildSearchEngineQueryFromSentence(String sentence) {
 				boolean bAccept = true;
 				for (String w : qs) {
 					if (w.toLowerCase().equals(w)) // idf only two words then
-						// has to be person name,
-						// title or geolocation
+					// has to be person name,
+					// title or geolocation
+					{
 						bAccept = false;
+						break;
```
[the remark this answers is not in the document]
Comment (rzo1) [reply opennlp-similarity/src/main/java/opennlp/tools/similarity/apps/ContentGenerator.java:153]: Missed the outer loop in diff view. Makes sense again, so nevermind :-)

**Review by rzo1 — APPROVED** [review]

File: opennlp-dl/src/main/java/opennlp/tools/dl/NeuralDocCatTrainer.java [hunk 4]
```diff
@@ -55,24 +55,24 @@ public class NeuralDocCatTrainer {
     public static class Args {
 
         @Option(name = "-batchSize", usage = "Number of examples in minibatch")
-        int batchSize = 128;
+        final int batchSize = 128;
```
[the remark this answers is not in the document]
Comment (mawiesne) [reply opennlp-dl/src/main/java/opennlp/tools/dl/NeuralDocCatTrainer.java:58]: That's an important hint! Will adjust and leave a comment here.

**Review by rzo1 — APPROVED** [review]
lgtm. thx.

**Pull request merged**